### DISCUSSION TITLE
What's wrong  with my code

### DISCUSSION THREAD

**Discussion by faridaf97** [quasarframework/quasar#13434] [Quasar App Extensions]
sorry i' changing my mind with the question haha, i'm very new in all this and I'm determined to learn it, but somehow don't always find the solution and don't know what to search.

I now have this code, where a menu is created. But it keeps opening in a new tab, which is not bad in a few menu points, but how do I keep some of them opening in the same tab and some in a new?

<template>
  <q-layout view="lHh Lpr lFf">
    <q-header elevated>
      <q-toolbar>
        <q-btn
          flat
          dense
          round
          icon="menu"
          aria-label="Menu"
          @click="toggleLeftDrawer"
        />

        <q-toolbar-title>
          Quasar App
        </q-toolbar-title>

        <div>Quasar v{{ $q.version }}</div>
      </q-toolbar>
    </q-header>

    <q-drawer
      v-model="leftDrawerOpen"
      show-if-above
      bordered
    >
      <q-list>
        <q-item-label
          header
        >
          Navigation
          
        </q-item-label>

        <EssentialLink
          v-for="link in essentialLinks"
          :key="link.title"
          v-bind="link"
        />
      </q-list>
    </q-drawer>

    <q-page-container>
      <router-view />
    </q-page-container>
  </q-layout>
</template>

<script>
import { defineComponent, ref } from 'vue'
import EssentialLink from 'components/EssentialLink.vue'

const linksList = [
  {
    title: 'Settings',
    caption: 'Change your personal settings',
    icon: 'settings',
    link: 'http://localhost:8080/#/settings'
  },
  {
   title: 'Self Help',
    caption: 'Find some informations about your condition here or some exercises that may help you',
    icon: 'psychology_alt',
    link: 'http://localhost:8080/#/selbsthilfe'
  },
  {
   title: 'Chat',
    caption: 'Chat with other people with your same problems',
    icon: 'chat',
    link: 'http://localhost:8080/#/chat'
  },
{
   title: 'Links',
    caption: 'Find help in your area here',
    icon: 'link',
    link: 'http://localhost:8080/#/links'
  },

  {
   title: 'Book Sessions',
    caption: 'Book sessions anonymously via "Doodle" through this button.',
    icon: 'event',
    link: 'xxxxxxxxxxxxxxxxxx'
  },

  {
   title: 'Enter Session',
    caption: 'Enter session. You should have received a passcode via E-Mail',
    icon: 'login',
    link: ' xxxxxxxxxxxxxxxx'
  },
  {
   title: 'My Notes',
    caption: 'Add you everyday thoughts here for your next session',
    icon: 'history_edu',
    link: 'http://localhost:8080/#/notes'
  },
]

export default defineComponent({
  name: 'MainLayout',

  components: {
    EssentialLink
  },

  setup () {
    const leftDrawerOpen = ref(false)

    return {
      essentialLinks: linksList,
      leftDrawerOpen,
      toggleLeftDrawer () {
        leftDrawerOpen.value = !leftDrawerOpen.value
      }
    }
  }
})
</script>

**Comment by MilosPaunovic** [comment]
Provide code reproduction; this is to vague.

**Reply by faridaf97** [reply]
I've changed it so it now only opens in the same tab, but now the external links don't work: 

<template>
  <q-layout view="lHh Lpr lFf">
    <q-header elevated>
      <q-toolbar>
        <q-btn
          flat
          dense
          round
          icon="menu"
          aria-label="Menu"
          @click="toggleLeftDrawer"
        />

        <q-toolbar-title>
          Quasar App
        </q-toolbar-title>

        <div>Quasar v{{ $q.version }}</div>
      </q-toolbar>
    </q-header>

    <q-drawer
      v-model="leftDrawerOpen"
      show-if-above
      bordered
    >
      <q-list>
        <q-item-label
          header
        >
          Navigation
          
        </q-item-label>


        <EssentialLink
          v-for="link in essentialLinks"
          :key="link.title"
          v-bind="link"
        />
      </q-list>
    </q-drawer>

    <q-page-container>
      <router-view />
    </q-page-container>
  </q-layout>
</template>

<script>
import { defineComponent, ref } from 'vue'
import EssentialLink from 'components/EssentialLink.vue'

const linksList = [
  {
    title: 'Settings',
    caption: 'Change your personal settings',
    icon: 'settings',
    to:'settings'
  },
  {
   title: 'Self Help',
    caption: 'Find some informations about your condition here or some exercises that may help you',
    icon: 'psychology_alt',
    to:'/selbsthilfe'
  },
  {
   title: 'Chat',
    caption: 'Chat with other people with your same problems',
    icon: 'chat',
    to: '/chat'
  },
{
   title: 'Links',
    caption: 'Find help in your area here',
    icon: 'link',
    to: '/links'
  },

  {
   title: 'Book Sessions',
    caption: 'Book sessions anonymously via "Doodle" through this button.',
    icon: 'event',
    link: ''
  },

  {
   title: 'Enter Session',
    caption: 'Enter session. You should have received a passcode via E-Mail',
    icon: 'login',
    link: ' '
  },
  {
   title: 'My Notes',
    caption: 'Add you everyday thoughts here for your next session',
    icon: 'history_edu',
    to: '/notes'
  },
]

export default defineComponent({
  name: 'MainLayout',

  components: {
    EssentialLink
  },

  setup () {
    const leftDrawerOpen = ref(false)

    return {
      essentialLinks: linksList,
      leftDrawerOpen,
      toggleLeftDrawer () {
        leftDrawerOpen.value = !leftDrawerOpen.value
      }
    }
  }
})
</script>

**Reply by faridaf97** [reply]
<template>
  <q-item
    clickable
    :to="to"
      >
    <q-item-section
      v-if="icon"
      avatar
    >
      <q-icon :name="icon" />
    </q-item-section>

    <q-item-section>
      <q-item-label>{{ title }}</q-item-label>
      <q-item-label caption>{{ caption }}</q-item-label>
    </q-item-section>
  </q-item>
</template>

<script>
import { defineComponent } from 'vue'

export default defineComponent({
  name: 'EssentialLink',
  props: {
    title: {
      type: String,
      required: true
    },

    caption: {
      type: String,
      default: ''
    },

    to: {
      type: String,
      default: '#'
    },

    icon: {
      type: String,
      default: ''
    }
  }
})
</script>

**Reply by MilosPaunovic** [reply]
This is not a proper reproduction. Please use either CodePen, CodeSandox or create a public repository with minimal reproduction. But first, do try to solve the issues by take a deep dive into both Vue and Quasar documentation.
